File: week8/PureAbs.java
interface Shape {
    double calculateArea();
}
class Circle implements Shape {
    private double radius;
    public Circle(double radius)
    {
        this.radius = radius;
    }
    public double calculateArea()
    {
        return Math.PI * radius * radius;
    }
}
class Rectangle implements Shape {
    private double length;
    private double width;
    public Rectangle(double length, double width)
    {
        this.length = length;
        this.width = width;
    }
    public double calculateArea() { return length * width; }
}
public class abstraction {
    public static void main(String[] args)
    {
        Circle c = new Circle(6.0);
        Rectangle r = new Rectangle(2.0, 4.0);
        System.out.println("Area of the Circle is : "+ c.calculateArea());
        System.out.println("Area of the Rectangle is: "+ r.calculateArea());
    }
}

File: week8/Pureabs.cpp
#include <iostream>
using namespace std;
// Abstract class
class Shape {
    protected:
    float dimension;
    public:
    void getDimension() 
    {
        cin >> dimension;
    }
    // pure virtual Function
    virtual float calculateArea() = 0;
};
class Square : public Shape {
    public:
    float calculateArea() 
    {
        return dimension * dimension;
    }
};
class Circle : public Shape {
    public:
    float calculateArea() 
    {
        return 3.14 * dimension * dimension;
    }
};
int main() {
    Square s;
    Circle c;
    cout << "Enter the length of the square: "<<endl;
    s.getDimension();
    cout << "Area of square: " << s.calculateArea() << endl;
    cout << "Enter radius of the circle: "<<endl;
    c.getDimension();
    cout << "Area of circle: " << c.calculateArea() << endl;
}
